mozilla/bedrock
# Show an info panel beside a release note if it's marked as 'progressive roll out'

## One-line summary

This changeset supports showing a new 'Progressive roll out' badge/panel beside certain release notes that (will) have this parameter set via Nucleus

<img width="1503" alt="Screenshot 2023-11-16 at 13 54 02" src="https://github.com/mozilla/bedrock/assets/101457/8e71f490-2809-4d60-933a-bee55adb00d9">



## Significant changes and points to review

* There's new CSS and a new image, but the only way to see them is to tweak the database (because the data isn't yet flowing from Nucleus) - see **Testing** for tips

* I've used cursor:help and a title attr to explain what a progressive rollout is, as suggested in the [original ticket](https://github.com/mozilla/nucleus/issues/810#issuecomment-1793262153) -- is there a better/more preferred way? Are there a11y issues doing it this way?  

* I've had to add a new field to `Note` to take `progressive_rollout` from release-notes' JSON and make it available as a Python attribute. I can't see any tests where we test the actual output of the rendered HTML release notes page, but I've added a test just for the additions I've made to the template itself

## Issue / Bugzilla link

Resolves #13891 which is the Bedrock side of https://github.com/mozilla/nucleus/issues/810

## Testing

* Download and unzip [bedrock-with-progressive-rollout-faked-data.db.zip](https://github.com/mozilla/bedrock/files/13378226/bedrock-with-progressive-rollout-faked-data.db.zip), rename it to `bedrock.db` and swap it in for your current `bedrock/data/bedrock.db` file -- I've tweaked some release notes to have the expected/future schema, including `progressive_rollout` on a couple of things
* `npm start`
* Go to http://localhost:8000/en-US/firefox/119.0.1/releasenotes/ and you'll see two items with the Progressive Roll Out badge
